packages/x-data-grid-premium/src/models/dataGridPremiumProps.ts

```diff
@@ -169,6 +169,14 @@ export interface DataGridPremiumPropsWithoutDefaultValue<R extends GridValidRowM
    * @param {string} inProgress Indicates if the task is in progress.
    */
   onExcelExportStateChange?: (inProgress: 'pending' | 'finished') => void;
+  /**
+   * Callback fired before the clipboard paste operation starts.
+   * Use it to confirm or cancel the paste operation.
+   * @param {object} params Params passed to the callback.
+   * @param {string[][]} params.data The raw pasted data split by rows and cells.
+   * @returns {Promise<any> | void} A promise that resolves to confirm the paste operation, and rejects to cancel it.
+   */
+  onClipboardPasteBeforeStart?: (params: { data: string[][] }) => Promise<any> | void;
```

Comment:
Overall, I like the idea 👍🏻

I would prefer to keep the `onClipboardPasteBeforeStart` name (or `onBeforeClipboardPasteStart`) – to make it clear when it's called.

> If continue paste operation => Promise.resolve(pastedValue)

It would also make sense to support `Promise.resolve()` (without a value) to make it less error-prone.

What do you think?

Reply:
> It would also make sense to support Promise.resolve() (without a value) to make it less error-prone.

You mean both these cases will be the same:
`Promise.resolve()`
`Promise.resolve(pastedValue)`

Or do you mean not to expect a value at all, would it be helpful to support data cleanup operations for some users?

Reply:
Is it that we could already [support](https://stackblitz.com/edit/react-vcpcoq?file=Demo.tsx) cleanup using `splitClipboardPastedText`? 
In any case, feel free to proceed with the merge.

Reply:
> You mean both these cases will be the same:
> Promise.resolve()
> Promise.resolve(pastedValue)

Correct.
With `processRowUpdate`, sometimes people experience issues because they didn't return a value.
So we can treat the `undefined` value as if the same value was returned.

> I think it also doesn't make sense to use this function for the cleanup

Sorry, I'm a bit lost on this one – which function are you referring to?

Reply:
> Sorry, I'm a bit lost on this one – which function are you referring to?

What I meant is, since the name `onBeforeClipboardPasteStart` doesn't explain the clean-up, it's something you want to execute before clipboard paste start. All good 👍